src/types.ts

```diff
@@ -1,26 +1,90 @@
+import { TransactionResponse } from '@ethersproject/abstract-provider';
+import { ethers } from 'ethers';
+
+import { SupportedChainId } from './config';
 import { Asset, Transaction } from './gnosis-safe/types';
-import { Proposal, TokenMetaData, Vote, zDAO } from './snapshot-io/types';
+import {
+  Proposal,
+  ProposalDetail,
+  ProposalResult,
+  TokenMetaData,
+  Vote,
+} from './snapshot-io/types';
 
 export interface SnapshotConfig {
+  // uri to Snaphost Hub
   serviceUri: string;
-  ipfsUri: string;
+  // ipfs gateway
+  ipfsGateway: string;
 }
 
 export interface GnosisSafeConfig {
+  // uri to Gnosis Safe service
   serviceUri: string;
-  safeAddress: string;
+  // uri to Gnosis Safe gateway
   gateway: string;
 }
 
+export interface zNAConfig {
+  // address to zDAOCore contract
+  contract: string;
+  // web3 provider
+  provider: ethers.providers.Provider;
+}
+
 export interface Config {
   snapshot: SnapshotConfig;
   gnosisSafe: GnosisSafeConfig;
-  chainId: string;
+  zNA: zNAConfig;
+}
+
+export type zNA = string;
+export type zDAOId = string;
+
+export interface zDAO {
+  id: zDAOId; // Global zDAO identifier
+  zNA: zNA; // Linked zNA
+  // If not defined, title should be zNA by default
+  title?: string; // zDAO title
+  creator: string; // Creator wallet address
+  owners: string[]; // Owner wallet addresses
+  // avatar uri which starts with https schema
+  // The frontend should use default avatar image if not defined
+  avatar?: string; // Avatar uri (https link)
+  network: string; // Chain id
+  safeAddress: string; // Gnosis Safe address
+  votingToken: string; // Voting token address
+}
+
+export interface CreateZDAODto {
```

Comment:
Nit pick: This is a "params` object not a "Dto"

DTO = Data Transfer Object. You use Data Transfer Objects when working with external systems such as API's (and sometimes other SDK's)

Params = An object that exists as a way of grouping parameters into an object instead of traditional way